app/com/linkedin/drelephant/ElephantRunner.java

```diff
@@ -178,8 +178,11 @@ public void run() {
         result.save();
         long processingTime = System.currentTimeMillis() - analysisStartTimeMillis;
         logger.info(String.format("Analysis of %s took %sms", analysisName, processingTime));
-        MetricsController.setJobProcessingTime(processingTime);
-        MetricsController.markProcessedJobs();
+
+        if(MetricsController.isInitialized()) {
```

Comment:
Is it not possible to ensure MetricsController is initialized before we start this thread?